src/sonic-eventd/rsyslog_plugin/rsyslog_plugin.h

```diff
@@ -0,0 +1,43 @@
+#ifndef RSYSLOG_PLUGIN_H
+#define RSYSLOG_PLUGIN_H
+
+#include <string>
+#include <fstream>
+#include "syslog_parser.h"
+#include "common/logger.h"
+#include "common/events.h"
+
+using namespace std;
+using json = nlohmann::json;
+
+/**
+ * Rsyslog Plugin will utilize an instance of a syslog parser to read syslog messages from rsyslog.d and will continuously read from stdin
+ * A plugin instance is created for each container/host.
+ *
+ */
+
+class RsyslogPlugin {
+public:
+    RsyslogPlugin(SyslogParser* syslog_parser, string mod_name, string path);
+    void onMessage(string msg);
```

Comment:
Other than run,  everything else can be private.

Reply:
I exposed onMessage for testing purposes and onInit as mentioned below.